src/mono/mono/metadata/threads.c

```diff
@@ -3912,18 +3915,58 @@ mono_thread_init_apartment_state (void)
 		thread->apartment_state = ThreadApartmentState_Unknown;
 	}
 #endif
+
+#if TARGET_OSX || HOST_MACCAT || TARGET_IOS || TARGET_TVOS
```

Comment:
host/target confusion.

but also this can be simplified to 

```suggestion
#if defined(HOST_DARWIN)
```

(and in various other places)